src/net/mod.rs

```diff
@@ -131,7 +132,7 @@ pub async fn init() -> Result<(), Error> {
 
 	// Setup the graceful shutdown
 	let handle = Handle::new();
-	let shutdown_handler = graceful_shutdown(handle.clone());
+	let shutdown_handler = graceful_shutdown(ct, handle.clone());
```

Comment:
In another PR I'd like to move this to start.ts somewhere close to where we currently do `let ct = CancellationToken::new();`. WDYT?!